zivid_camera/src/zivid_camera.cpp

```diff
@@ -379,6 +366,72 @@ bool ZividCamera::capture2DServiceHandler(Capture::Request&, Capture::Response&)
   return true;
 }
 
+bool ZividCamera::captureAssistantSuggestSettingsServiceHandler(CaptureAssistantSuggestSettings::Request& req,
+                                                                CaptureAssistantSuggestSettings::Response&)
+{
+  ROS_DEBUG_STREAM(__func__ << ": Request: " << req);
+
+  serviceHandlerHandleCameraConnectionLoss();
+
+  const auto max_capture_time =
+      std::chrono::round<std::chrono::milliseconds>(std::chrono::duration<double>{ req.max_capture_time.toSec() });
+  const auto ambient_light_frequency = [&req]() {
+    switch (req.ambient_light_frequency)
+    {
+      case CaptureAssistantSuggestSettings::Request::AMBIENT_LIGHT_FREQUENCY_NONE:
+        return Zivid::CaptureAssistant::AmbientLightFrequency::none;
+      case CaptureAssistantSuggestSettings::Request::AMBIENT_LIGHT_FREQUENCY_50HZ:
+        return Zivid::CaptureAssistant::AmbientLightFrequency::hz50;
+      case CaptureAssistantSuggestSettings::Request::AMBIENT_LIGHT_FREQUENCY_60HZ:
+        return Zivid::CaptureAssistant::AmbientLightFrequency::hz60;
+    }
+    throw std::runtime_error("Unhandled AMBIENT_LIGHT_FREQUENCY value: " + std::to_string(req.ambient_light_frequency));
+  }();
+
+  Zivid::CaptureAssistant::SuggestSettingsParameters suggest_settings_parameters(max_capture_time,
+                                                                                 ambient_light_frequency);
+  ROS_INFO_STREAM("Getting suggested settings using parameters: " << suggest_settings_parameters);
+  const auto suggested_settings{ Zivid::CaptureAssistant::suggestSettings(camera_, suggest_settings_parameters) };
+
+  if (suggested_settings.empty())
+  {
+    throw std::runtime_error("The suggestSettings function returned 0 settings!");
+  }
+  if (suggested_settings.size() > capture_frame_config_dr_servers_.size())
+  {
+    throw std::runtime_error("The number of suggested settings (" + std::to_string(suggested_settings.size()) + ") " +
+                             "is larger than the number of dynamic_reconfigure capture/frame_<n> servers (" +
+                             std::to_string(capture_frame_config_dr_servers_.size()) + "). Increase launch parameter " +
+                             "num_capture_frames. See README.md for more information.");
+  }
+
+  ROS_INFO_STREAM("CaptureAssistant::suggestSettings returned " << suggested_settings.size() << " settings");
```

Comment:
Suggestion: Move this print to the sample, showing how to check what settings were suggested by the Capture Assistant

Reply:
Did you agree on whether or not to do this?

Reply:
This was discussed below, in sample_capture_assistant.cpp.